### ISSUE TITLE
mvnd site:run stays running after mvnd process stops

### ISSUE THREAD

**Issue by bdemers** [apache/maven-mvnd#1044]
The embedded Jetty instance stays running after the `mvnd` bin process is stopped.

Tested with versions: 1.0.0 and 1.0.1 (currently under vote)

Steps to repro:
Clone https://github.com/apache/maven-site

Run `mvnd site:run`

open `http://localhost:8080/` (everything works fine)
Back in your terminal kill the process: `^C`

Refresh your browser (pages still load)
run `mvnd --stop` to kill the embedded jetty process

**Comment by cstamas** [comment]
IMO this is more m-site-p bug than mvnd: the `site:run` does not offer any means to stop (in other words, it implicitly relies on the fact it is "local" in JVM and Ctrl+C stop it. It should probably have some prompt on console and once got input, cleanly stop?

**Comment by cstamas** [comment]
See https://github.com/apache/maven-site-plugin/blob/master/src/main/java/org/apache/maven/plugins/site/run/SiteRunMojo.java#L97-L102

**Comment by gnodet** [comment]
When the client is aborted using Ctrl+C, it sends a message to the server to stop the build.  I don't recall what the server does at this point, but if it's not done yet, maybe there's a way to interrupt the current threads.

**Comment by cstamas** [comment]
Yup, that kills probably the "main" thread (that sits to join on link above). But Jetty spawned already another non-daemon thread...
